crates/types/src/block_impl.rs

```diff
@@ -25,10 +25,35 @@ pub const NUM_STORAGE_NODES: usize = 8;
 /// Number of chunks for VID initiation.
 pub const NUM_CHUNKS: usize = 8;
 
+/// The error type for block and its transactions.
+#[derive(Snafu, Debug)]
+pub enum BlockError {
+    /// Invalid block header.
+    InvalidBlockHeader,
+}
+
 /// The transaction in a [`VIDBlockPayload`].
 #[derive(Default, PartialEq, Eq, Hash, Serialize, Deserialize, Clone, Debug)]
 pub struct VIDTransaction(pub Vec<u8>);
 
+impl VIDTransaction {
+    #[must_use]
+    /// Encode a list of transactions into bytes.
+    pub fn encode(transactions: Vec<Self>) -> Vec<u8> {
+        let mut encoded = Vec::new();
+
+        for txn in transactions {
+            // Encode the length of the inner transaction and the transaction bytes.
+            if let Ok(len) = txn.0.len().try_into() {
```

Comment:
Since we're converting the length to a `u8`, this means we can't have any transactions bigger than 256 bytes. Would this be a problem for benchmarking? If so I think you can take the length as a `u64` or `u32` and use `to_le_bytes()` to get a binary representation of the length.

Also, probably not good to silently skip transactions here when the length is too large. Better to panic or at least log something